src/provision/freedom-module.js

```diff
@@ -0,0 +1,351 @@
+//var XMLHttpRequest = require('freedom-xhr').corexhr;
+// Included directly in freedom module. node-forge not browserify-friendly
+//var forge = require('node-forge')({ disableNativeCode: true });
+
+var POLL_TIMEOUT = 5000; //milliseconds
+
+var STATUS_CODES = {
+  "START": "Starting provisioner",
+  "OAUTH_INIT": "Initializing oauth flow",
+  "OAUTH_ERROR": "Error getting oauth token",
+  "OAUTH_COMPLETE": "Got oauth token",
+  "SSHKEY_RETRIEVED": "Retrieved SSH keys from storage",
+  "SSHKEY_GENERATED": "Generated new SSH keys",
+  "CLOUD_FAILED": "Failed to complete cloud operation",
+  "CLOUD_INIT_ADDKEY": "Starting to add SSH key to cloud account",
+  "CLOUD_DONE_ADDKEY": "Done adding SSH key to cloud account",
+  "CLOUD_INIT_VM": "Starting to provision VM",
+  "CLOUD_DONE_VM": "Done provisioning VMs",
+  "CLOUD_WAITING_VM": "Waiting on VM",
+};
+
+var REDIRECT_URIS = [
+  "https://kmnjcbmibpajfljogomikcdlmhgpnolg.chromiumapp.org"
+  //  "http://localhost:10101"
+];
+
+var Provisioner = function(dispatchEvent) {
+  this.dispatch = dispatchEvent;
+  this.state = {};
+};
+
+/**
+ * Dispatches status events
+ * events listed in STATUS_CODES
+ * @param {String} code - one of STATUS_CODES 
+ **/
+Provisioner.prototype._sendStatus = function(code) {
+  this.dispatch("status", {
+    "code": code,
+    "message": STATUS_CODES[code]
+  });
+};
+
+/*
+ * Generates an RSA keypair using forge
+ * @return {Object.<String,String>} public and private SSH keys
+ */
+Provisioner.prototype._generateKeyPair = function() {
+  "use strict";
+  var pair = forge.pki.rsa.generateKeyPair({bits: 1024, e: 0x10001});  // TODO: use async
```

Comment:
Ah good catch. This was copied from Colony, which I think lowered it because it was running in the main thread, which blocked everything else. Now that it's in a freedom.js module, we shouldn't worry about blocking the thread as much.